geopandas/io/sql.py

```diff
@@ -96,3 +96,222 @@ def load_geom_text(x):
                 crs = {"init": "epsg:{}".format(srid)}
 
     return GeoDataFrame(df, crs=crs, geometry=geom_col)
+
+
+def get_geometry_type(gdf):
+    """
+    Get basic geometry type of a GeoDataFrame,
+    and information if the gdf contains Geometry Collections."""
+    geom_types = list(gdf.geometry.geom_type.unique())
+    geom_collection = False
+
+    # Get the basic geometry type
+    basic_types = []
+    for gt in geom_types:
+        if "Multi" in gt:
+            geom_collection = True
```

Comment:
This does not check for GeometryCollection but Multi-types only. 

Btw, what would happen with GeometryCollections? Are they somehow supported?

Reply:
Yep they indeed are supported, learned just now after your question (thanks!). I will add handling of those to here as well.

Reply:
Added now support for GeometryCollections, as well as for LinearRings (which are converted to LineStrings).